### PR TITLE
new component EditSpaceModal

### PR DESCRIPTION
**Pull request by TyaglovOV**
all types will be moved to web-types later.

### CODE REVIEW

File: src/components/EditSpaceModal/EditSpaceModal.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+      screen: 'info',
+      space: {
+        name: props.space.name,
+        shortname: props.space.shortname,
+        avatar: props.space.avatar
+      }
+    };
+  }
+
+  handleChange = (value: mixed, { target }: $FlowIssue) => {
```
Comment (meafmira) [inline src/components/EditSpaceModal/EditSpaceModal.js]: Why `$FlowIssue`

File: src/components/EditSpaceModal/EditSpaceModal.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+import type { Props, State } from './types';
+import HotKeys from '../HotKeys/HotKeys';
+
+class EditSpaceModal extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      screen: 'info',
+      space: {
```
Comment (meafmira) [inline src/components/EditSpaceModal/EditSpaceModal.js]: Use `getDerivedStateFromProps` for this

File: src/components/EditSpaceModal/EditSpaceModalForm.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+class EditSpaceModalForm extends PureComponent<Props, State> {
+  static contextTypes = {
+    l10n: LocalizationContextType
+  };
+
+  constructor(props: Props, context: Context) {
+    super(props, context);
+
+    this.state = {
```
Comment (meafmira) [inline src/components/EditSpaceModal/EditSpaceModalForm.js]: Use `getDerivedStateFromProps`

File: src/components/EditSpaceModal/EditSpaceModal.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        onChange={this.handleChange}
+        onSubmit={this.handleSubmit}
+        onAvatarChange={this.handleAvatarEdit}
+        onAvatarRemove={this.handleAvatarRemove}
+      />
+    );
+  }
+
+  renderAvatarEdit() {
+    if (!this.state.space.avatar || typeof this.state.space.avatar === 'string') {
```
Comment (meafmira) [inline src/components/EditSpaceModal/EditSpaceModal.js]: Don't use `typeof` anywhere. Use Flow for typings

File: src/components/EditSpaceModal/EditSpaceModalForm.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      avatar: props.avatar
+    };
+
+    if (props.avatar && typeof props.avatar !== 'string') {
```
Comment (meafmira) [inline src/components/EditSpaceModal/EditSpaceModalForm.js]: Don't use typeof anywhere. Use Flow for typings

File: src/components/EditSpaceModal/EditSpaceModalForm.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      fileToBase64(props.avatar, (avatar) => {
+        this.setState({ avatar });
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.avatar !== this.props.avatar &&
+      typeof this.props.avatar !== 'string' &&
```
Comment (meafmira) [inline src/components/EditSpaceModal/EditSpaceModalForm.js]: Don't use typeof anywhere. Use Flow for typings

File: src/components/EditSpaceModal/EditSpaceModalForm.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    }
+  }
+
+  handleSubmit = (event: SyntheticEvent<>) => {
+    event.preventDefault();
+
+    this.props.onSubmit();
+  };
+
+  getInputState = (field: string): Object => {
```
Comment (meafmira) [inline src/components/EditSpaceModal/EditSpaceModalForm.js]: Use enum with input names instead of `string`

**Comment by meafmira** [comment]
@TyaglovOV there's flow error - Don't use typeof anywhere. Use Flow for typings

File: src/components/EditSpaceModal/EditSpaceModalForm.js (hunk 7 above)
Comment (TyaglovOV) [reply src/components/EditSpaceModal/EditSpaceModalForm.js]: done

**Pull request merged**